splinterd/tests/admin/node_lifecycle.rs

```diff
@@ -0,0 +1,161 @@
+///    retrieved from the network
+/// 8. Start the first node and check that it can now be retrieved from the network
+/// 9. Check that the first node is running by getting its registry client and using it
+/// 10. Check that the second and thrid nodes are still stopped
+/// 11. Start the second node and check that it can now be retrieved from the network
+/// 12. Check that the second node is running by getting its registry client and using it
+/// 13. Check that the third node is still stopped
+/// 14. Start the third node and check that it can now be retrieved from the network
+/// 15. Check that the third node is running by getting its registry client and using it
+/// 16. Shutdown
+#[test]
+pub fn test_stop_start_nodes_multiple() {
+    // create network with three nodes
+    let mut network = Network::new()
+        .with_default_rest_api_variant(RestApiVariant::ActixWeb1)
+        .add_nodes_with_defaults(3)
+        .expect("Unable to start single node ActixWeb1 network");
+
+    // Stop the first node
+    network = network.stop(0).expect("Unable to stop first node");
+
+    // Check that the first node is stopped and can not be retrieved from the network
+    assert!(network.node(0).is_err());
+    // Check that the second and third nodes are still running and can be retrieved from
+    // the network
+    let node_1 = network.node(1).expect("Unable to retrieve second node");
+    let node_2 = network.node(2).expect("Unable to retrieve third node");
+    // Test that the second and third nodes are running by using the registry client
+    let registry_response = node_1
+        .registry_client()
+        .get_node(node_1.node_id())
+        .expect("Unable to list nodes from registry");
+    assert_eq!(node_1.node_id(), registry_response.unwrap().identity);
+    let registry_response = node_2
+        .registry_client()
+        .list_nodes(None)
+        .expect("Unable to list nodes from registry");
+    assert_eq!(3, registry_response.data.len());
+
+    // Stop the second node
+    network = network.stop(1).expect("Unable to stop the second node");
+
+    // Check that the second node is stopped and can not be retrieved from the network
+    assert!(network.node(1).is_err());
+    // Check that the first node is still stopped and can not be retrieved from
+    // the network
+    assert!(network.node(0).is_err());
+    // Check that the third nodes is still running and can be retrieved from the network
+    let node_2 = network.node(2).expect("Unable to retrieve third node");
+    // Test that the third node is running by using the registry client
+    let registry_response = node_2
+        .registry_client()
+        .get_node(node_2.node_id())
+        .expect("Unable to list nodes from registry");
+    assert_eq!(
+        node_2.network_endpoints(),
+        registry_response.unwrap().endpoints
+    );
+
+    // Stop the third node
+    network = network.stop(2).expect("Unable to stop the second node");
```

Comment:
Copy pasta in the expect message here again. This comment applies to the following expects as well, I noticed a few other expect messages that don't pertain to the thing being expected (below this line in this file).

Reply:
Fixed